src/main/java/com/redhat/devtools/lsp4ij/client/features/LSPEditorBehaviorFeature.java

```diff
@@ -0,0 +1,132 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at https://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+
+package com.redhat.devtools.lsp4ij.client.features;
+
+import com.intellij.psi.PsiFile;
+import com.redhat.devtools.lsp4ij.LanguageServiceAccessor;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * LSP client-side editor behavior feature. This does not correspond to an actual LSP feature.
+ */
+@ApiStatus.Experimental
+public class LSPEditorBehaviorFeature {
```

Comment:
The client-side editor behavior feature itself with flags for each of the editor improvement features that I've added in this PR. These flags all default to enabled as I think most(/all) users will want them for the behavioral parity they provide with natively-supported languages, but if there are issues -- or if a user just doesn't like how one or more of these behave -- they can be disabled via client config.